src/lib/components/drs/DrsMain.js

```diff
@@ -216,18 +215,29 @@ const DrsMain = props => {
     parents: _.pick(formProps, ['drs_object_parents', 'addParent', 'setParentId', 'setParentName', 'setParentValid', 'setParentInvalid', 'unsetParentValidity', 'removeParent']),
     fileAccessObjects: _.pick(formProps, ['file_access_objects', 'addFileAccessObject', 'setFileAccessObjectPath', 'removeFileAccessObject']),
     awsS3AccessObjects: _.pick(formProps, ['aws_s3_access_objects', 'addAwsS3AccessObject', 'setAwsS3AccessObjectRegion', 'setAwsS3AccessObjectBucket', 'setAwsS3AccessObjectKey', 'removeAwsS3AccessObject']),
-    submit: {activeDrsObject: activeDrsObject, setSuccessMessage: setSuccessMessage, ..._.pick(formProps, ['retrieveDrsObjectsList'])}
+    submit: {activeDrsObject: activeDrsObject, setSuccessMessage: setSuccessMessage, ..._.pick(formProps, ['retrieveDrsObjectsList'])}, 
+    edit: _.pick(formProps, ['id']), 
+    cancel: _.pick(formProps, ['id']),
+    delete: {setSuccessMessage: setSuccessMessage, ..._.pick(formProps, ['id', 'retrieveDrsObjectsList'])}
   }
 
   /* ##################################################
    * EFFECTS
    * ################################################## */
 
   const loadActiveDrsObjectBasedOnPath = location => {
-    let re = new RegExp('/drs/(.+)');
+    let re = new RegExp('/drs/(.+[^/edit])');
```

Comment:
this regex pattern is not correctly configured. It is saying "match all the characters except the final character, which cannot be a slash, 'e', 'd', 'i', or 't'". This won't behave correctly for many ids.

Instead we want to say, "match everything after '/drs/', but not a slash and not anything after the slash if one is found".

try this: `'/drs/([^/]+)'` 

we should not need more than one regex pattern once it's working correctly